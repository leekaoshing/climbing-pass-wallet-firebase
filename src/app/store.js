import { configureStore } from '@reduxjs/toolkit';
import { firebaseReducer } from 'react-redux-firebase';
import { actionTypes, firestoreReducer } from 'redux-firestore';
import userReducer from '../reducers/userSlice';
import dialogReducer from '../reducers/dialogSlice';

// import { createStore, applyMiddleware, compose } from 'redux';
// import reduxThunk from 'redux-thunk';
// import { ReactReduxFirebaseProvider, getFirebase } from 'react-redux-firebase'
// import { reduxFirestore, getFirestore } from 'redux-firestore';
// import { devToolsEnhancer, composeWithDevTools } from 'redux-devtools-extension';
// import thunk from 'redux-thunk';
// import firebase from '../services/firebase';

export const rrfConfig = {
  userProfile: 'users',
  attachAuthIsReady: true,
  useFirestoreForProfile: true,
  onAuthStateChanged: (authData, firebase, dispatch) => {
    // Clear redux-firestore state if auth does not exist (i.e logout)
    if (!authData) {
      dispatch({ type: actionTypes.CLEAR_DATA })
    }
 }
}

// export default configureStore = () => {
//   const middlewares = [thunk.withExtraArgument(getFirebase)];

//   const composedEnhancer = composeWithDevTools(
//     applyMiddleware(...middlewares),
//     reactReduxFirebase(firebase, rrfConfig),
//     reduxFirestore(firebase)
//   );

//   const store = createStore({
//     user: userReducer,
//     gym: gymReducer
//   }, composedEnhancer)

//   return store
// }

// const rootReducer = combineReducers({
//   user: userReducer,
//   gym: gymReducer,
//   firebase: firebaseReducer,
//   firestore: firestoreReducer
// })

// // Create store with reducers and initial state
// const initialState = {}
// export default createStore(rootReducer, initialState);

export const rootReducer = {
  user: userReducer,
  dialog: dialogReducer,
  firebase: firebaseReducer,
  firestore: firestoreReducer
};

export default configureStore({
  reducer: rootReducer
});
