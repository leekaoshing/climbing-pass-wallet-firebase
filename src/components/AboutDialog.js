import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
import { useDispatch, useSelector } from 'react-redux';
import {
    selectShowAboutDialog,
    setShowAboutDialog
} from '../reducers/userSlice';

const useStyles = makeStyles((theme) => ({
    paper: {
        padding: theme.spacing(3),
        textAlign: 'center',
    }
}));

export function AboutDialog() {
    const dispatch = useDispatch();
    const classes = useStyles();
    const showAboutDialog = useSelector(selectShowAboutDialog);

    const handleClose = () => {
        dispatch(setShowAboutDialog(false));
    }

    return (
        <Dialog onClose={handleClose} aria-labelledby="about-dialog-title" open={showAboutDialog}>
            <DialogTitle>About</DialogTitle>
            <Paper className={classes.paper} elevation={0}>
                <p style={{ textAlign: 'justify' }}>
                    Welcome to the <b>Climbing Pass Wallet</b>! This is a community-driven project for climbers to keep track of their existing multipasses.
                    No more calling up gyms or logging into different websites just to check your remaining passes :)
                </p>
                <p style={{ textAlign: 'justify' }}>
                    At this stage, this wallet will require manual recording of passes. Simply create an account if you haven't already, or log in,
                    and add or subtract passes as you buy more passes or visit gyms.
                </p>
                <p style={{ textAlign: 'justify' }}>
                    If there is sufficient community adoption, our next step would be to approach the various gyms to link this to their databases for
                    automatic updating of passes. Do share this site with your friends to help out!
                </p>
            </Paper>
            <DialogActions>
                <Button
                    color="primary"
                    onClick={handleClose}
                >
                    Close
                </Button>
            </DialogActions>
        </Dialog>
    );
}