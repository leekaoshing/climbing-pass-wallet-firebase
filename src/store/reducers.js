import { combineReducers } from 'redux'
import firebase from 'react-redux-firebase/lib/reducer'
import firestore from 'redux-firestore/lib/reducer'
import locationReducer from './reducers/location'
import dialogReducer from './reducers/dialog'
import userReducer from './reducers/user'

export function makeRootReducer(asyncReducers) {
  return combineReducers({
    // Add sync reducers here
    firebase,
    firestore,
    dialog: dialogReducer,
    user: userReducer,
    location: locationReducer,
    ...asyncReducers
  })
}

export function injectReducer(store, { key, reducer }) {
  store.asyncReducers[key] = reducer // eslint-disable-line no-param-reassign
  store.replaceReducer(makeRootReducer(store.asyncReducers))
}

export default makeRootReducer
