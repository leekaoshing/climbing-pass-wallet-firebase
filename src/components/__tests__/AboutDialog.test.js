import {
    fireEvent, render,
    screen
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../app/store';
import { AboutDialog } from '../AboutDialog';

jest.disableAutomock()

describe.skip('About Dialog', () => { // TODO Material-UI Dialog doesn't render snapshots properly
    it('renders and performs close action correctly', async () => {
        const { asFragment } = render(
            <Provider store={store}>
                <AboutDialog />
            </Provider>,
        );

        expect(asFragment()).toMatchSnapshot();
        const button = await screen.findAllByText(/About/i);
        fireEvent.click(button);

        expect(selectShowAboutDialog(store.getState())).toBe(false);
    });
})