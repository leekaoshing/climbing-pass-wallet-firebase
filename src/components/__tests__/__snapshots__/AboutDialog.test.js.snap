// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Dialog renders and performs close action correctly 1`] = `<DocumentFragment />`;
